x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/LifecyclePolicyTests.java

```diff
@@ -145,6 +146,24 @@ public static LifecyclePolicy randomTimeseriesLifecyclePolicyWithAllPhases(@Null
     }
 
     public static LifecyclePolicy randomTimeseriesLifecyclePolicy(@Nullable String lifecycleName) {
+        LifecyclePolicy policy = null;
+
+        // Keep generating a random policy until it passes the validation,
+        // ignoring known specific invalid states
+        while (policy == null) {
```

Comment:
Would it be a bit faster to validate when generating the random policy that if we added `forcemerge` we will add `rollover`?

Reply:
+1 for @andreidan idea, currently we risk potentially very long running tests (or random timeouts) if we can't generate correct policy